timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -282,6 +285,30 @@ public void multipleLockRequestsWithTheSameIdAreGranted() {
         }
     }
 
+    @Test
+    @Ignore // TODO (jkong): Fix this test by reworking the threading model.
+    public void stressTest() {
+        TestableTimelockServer nonLeader = Iterables.getFirst(cluster.nonLeaders(client.namespace()).values(), null);
+        int startingNumThreads = ManagementFactory.getThreadMXBean().getThreadCount();
+        try {
+            for (int i = 0; i < 10_000; i++) { // Needed as it takes a while for the thread buildup to occur
+                client.getFreshTimestamp();
+                assertThat(ManagementFactory.getThreadMXBean().getThreadCount())
+                        .as("should not additionally spin up too many threads")
+                        .isLessThanOrEqualTo(startingNumThreads + 200);
+                if (i == 1_000) {
+                    nonLeader.serverHolder().wireMock().register(
```

Comment:
Extract a method indicating that this simulates the node shutting down?

Reply:
Will do!